training_controller/prepare_training_logs.py

```diff
@@ -175,78 +147,126 @@ def fetch_training_logs(self, es_instance, num_logs_to_fetch, timestamps_list):
+                os.path.join(self.ES_DUMP_DIR, f"{filename}")
             )
             query_queue.append(current_command)
         if len(query_queue) > 0:
             self.run_esdump(query_queue)
+            return True
+        else:
+            return False
 
-    def create_windows(self):
-        # For every json file, write/append each time window to own file
-        for es_split_json_file in os.listdir(self.es_dump_data_path):
-            if not ".json" in es_split_json_file:
-                continue
-            json_file_to_process = os.path.join(
-                self.es_dump_data_path, es_split_json_file
-            )
-            df = pd.read_json(json_file_to_process, lines=True)
-            df = pd.json_normalize(df["_source"])
-            df.groupby(["window_start_time_ns"]).apply(
-                lambda x: self.save_window(x.name, x)
-            )
-            # delete ESDumped file
-            os.remove(json_file_to_process)
-        shutil.rmtree(self.es_dump_data_path)
 
     def fetch_and_update_timestamps(self,es_instance):
         timestamps_list = []
         try:
             oldest_log = es_instance.search(index="logs", body={"aggs": {"min_ts": {"min": { "field": "timestamp"}}}, "_source": ["timestamp"]}, size=1)
+            newest_log = es_instance.search(index="logs", body={"aggs": {"max_ts": {"max": { "field": "timestamp"}}}, "_source": ["timestamp"]}, size=1)
         except Exception as e:
             logging.error(e)
             return timestamps_list
         oldest_log_timestamp = int(oldest_log["aggregations"]["min_ts"]["value"])
+        newest_log_timestamp = int(newest_log["aggregations"]["max_ts"]["value"])
         try:
             all_normal_intervals = scan(es_instance, index="opni-normal-intervals", query={"query": {"match_all": {}}})
         except Exception as e:
             logging.error("Error trying to retrieve all normal intervals from opni-normal-intervals index")
             return timestamps_list
         for normal_interval in all_normal_intervals:
             start_ts, end_ts = normal_interval["_source"]["start_ts"], normal_interval["_source"]["end_ts"]
+            full_file_prefix = "{}_{}".format(start_ts, end_ts)
+            interval_json_files = [
+                file
+                for file in os.listdir(self.TRAINING_DIR)
+                if full_file_prefix in file
+            ]
             if end_ts < oldest_log_timestamp:
                 try:
                     es_instance.delete(index="opni-normal-intervals", doc_type=normal_interval["_type"], id=normal_interval["_id"])
                     logging.info("Deleting old normal time interval from Elasticsearch")
+                    for interval_file in interval_json_files:
+                        os.remove(os.path.join(self.TRAINING_DIR,interval_file))
+                        logging.info("Removing old JSON training files where the time interval within opni-normal-intervals was deleted.")
```

Comment:
is this intended?
just because end_ts is less than oldest_log_timestamp doesn't mean all the intervals in between fall into this condition

Reply:
thus those intervals shouldn't be deleted

Reply:
interval_json_files is a list which contains all the files within the self.TRAINING_DIR directory that start with the prefix related to the start_ts and end_ts. Thus, only the files where the end_ts is before the oldest_log_timestamp will be deleted not every interval.